src/utils/twoFactor.js

```
@@ -0,0 +1,181 @@
+}
+const addRequestAndConfirm = async (contract, request) => {
+    return contract.add_request_and_confirm({ request }).catch((e) => console.log(e))
+}
+const getContract = (account) => {
+    return new nearApiJs.Contract(account, account.accountId, {
+        viewMethods: VIEW_METHODS,
+        changeMethods: METHOD_NAMES_LAK,
+        sender: account.accountId
+    });
+}
+/********************************
+Exports
+********************************/
+export const addKeyAction = (account, publicKey, contractId, fullAccess = false) => {
+    const { accountId } = account
+    if (!contractId) contractId = accountId
+    let allowance = ACCESS_KEY_FUNDING_AMOUNT
+    let method_names = METHOD_NAMES_LAK
+    if (contractId !== accountId) {
+        fullAccess = false
+        method_names = ['']
+    } else {
+        allowance = '0'
+    }
+    return {
+        type: 'AddKey',
+        public_key: splitPK(publicKey),
+        ...(!fullAccess ? {
+            permission: {
+                receiver_id: contractId,
+                allowance,
+                method_names
+            }
+        } : null)
+    }
+}
+
+export const deleteKeyAction = (publicKey) => ({ type: 'DeleteKey', public_key: splitPK(publicKey) })
+
+export const getRequest = () => {
+    return JSON.parse(localStorage.getItem(`__multisigRequest`) || `{}`)
+}
+
+export const setRequest = (data) => {
+    localStorage.setItem(`__multisigRequest`, JSON.stringify(data))
+}
+
+export const sendTwoFactorRequest = async (wallet, accountId, method, requestId = -1, data = {}) =>  {
+    if (!accountId) accountId = wallet.accountId
+    if (!method) method = await wallet.get2faMethod()
+    // add request to local storage
+    setRequest({ accountId, requestId, data })
+
+    const serverResponse = wallet.postSignedJson('/2fa/send', {
+        accountId,
+        method,
+        requestId,
+        data
+    }).catch((e) => console.log('/2fa/send failed', e));
```

why `.catch` in `async` function? Why ignore error?

---

updated to try, catch

reorg function to return early